File: StaticLib1/Symbols.cpp
#include "pch.h"
#include "Symbols.h"
Symbols::Symbols(const string& v) : ryad(v) {}

string Symbols::replace() const
{
	string res;
	for (int i = 0; i < value.length(); i++)
	{
		if (value[i] == '4')
		{
			res += "!!";
		}
		else
			res += value[i];
	}
	return res;
}

File: ClassLibrary1/Symbols.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClassLibrary1
{
    public class Symbols : Class1
    {
        public Symbols(string v) : base(v)
        {
        }
        public override string Replace()
        {
            string res = "";
            foreach (char c in value)
            {
                if (c == '4')
                    res += "!!";
                else
                    res += c;
            }
            return res;
        }
    }
}